core/src/consensus.rs

```diff
@@ -265,45 +265,45 @@ impl Tower {
 
     fn new_vote(
         local_vote_state: &VoteState,
-        slot: u64,
+        slot: Slot,
         hash: Hash,
-        last_bank_slot: Option<Slot>,
+        last_voted_slot_in_bank: Option<Slot>,
     ) -> (Vote, usize) {
         let mut local_vote_state = local_vote_state.clone();
         let vote = Vote::new(vec![slot], hash);
         local_vote_state.process_vote_unchecked(&vote);
-        let slots = if let Some(last_bank_slot) = last_bank_slot {
+        let slots = if let Some(last_voted_slot_in_bank) = last_voted_slot_in_bank {
             local_vote_state
                 .votes
                 .iter()
                 .map(|v| v.slot)
-                .skip_while(|s| *s <= last_bank_slot)
+                .skip_while(|s| *s <= last_voted_slot_in_bank)
                 .collect()
         } else {
             local_vote_state.votes.iter().map(|v| v.slot).collect()
         };
         trace!(
             "new vote with {:?} {:?} {:?}",
-            last_bank_slot,
+            last_voted_slot_in_bank,
             slots,
             local_vote_state.votes
         );
         (Vote::new(slots, hash), local_vote_state.votes.len() - 1)
     }
 
-    fn last_bank_vote(bank: &Bank, vote_account_pubkey: &Pubkey) -> Option<Slot> {
+    fn last_voted_slot_in_bank(bank: &Bank, vote_account_pubkey: &Pubkey) -> Option<Slot> {
         let vote_account = bank.vote_accounts().get(vote_account_pubkey)?.1.clone();
         let bank_vote_state = VoteState::deserialize(&vote_account.data).ok()?;
-        bank_vote_state.votes.iter().map(|v| v.slot).last()
```

Comment:
I don't know how much rustc optimizer is clever. But I simply changed the algorithm from map-then-last to last-then-map, considering this would be more straight-forward in this case: https://github.com/solana-labs/solana/pull/10747/files#diff-3666262446ea129567c6d96960c7e23cR411-R418

New code might eliminate this unneeded iteration here.